aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp

```diff
@@ -21,16 +22,43 @@
 namespace at {
 namespace native {
 namespace {
-// FIXME: make this thread-safe by reusing the benchmark cache in Conv_v7.cpp
-namespace {
+constexpr uint8_t max_num_input_dim = 5;
+struct AddParams {
+  c10::DeviceIndex device_id;
+  int input_a_size[max_num_input_dim];
+  int input_b_size[max_num_input_dim];
+  uint8_t input_dim; // we currently assume both inputs are given as the same size (i.e., no broadcasting)
+  at::MemoryFormat memory_format;
+  bool deterministic;
+  bool allow_tf32;
+};
 struct CacheKey {
+  AddParams params;
   uint8_t input_a_alignment;
   uint8_t input_b_alignment;
   uint8_t output_alignment;
   bool kReluFused;
 };
-std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, at::native::ParamsHash<CacheKey>, at::native::ParamsEqual<CacheKey>> execution_plan_cache;
+void setAddParams(
+    AddParams* params, const at::Tensor& input_a, const at::Tensor& input_b,
+    bool deterministic, bool allow_tf32) {
+  // operator datatype needs to be int32 for int8 matmul, but we can
+  // set the datatype for output tensor to int32 or fp32
+  memset(params, 0, sizeof(AddParams));
```

Comment:
do we need to do this?

Reply:
I copied that over from https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cudnn/ConvShared.cpp#L89. 
I think it's used for initialization and to set the remaining values of `input_*_size` to zero. Without it, I think we'd just need an additional loop to initialize the remaining values.